### ISSUE TITLE
How to change the colors of console?

### ISSUE THREAD

**Issue by ok97465** [spyder-ide/spyder#16607]
### What steps will reproduce the problem?

Hello. I want the colors of the console.
(Prompt, Error message, ...)
How can I change the colors?

![console](https://user-images.githubusercontent.com/22970578/137747033-eb989560-f7a8-42b9-a197-116a8c01f6d4.png)

#Spyder version: 5.2.0.dev0(#3cdcfc13e)

# Mandatory:
atomicwrites >=1.2.0                  :  1.4.0 (OK)
chardet >=2.0.0                       :  4.0.0 (OK)
cloudpickle >=0.5.0                   :  2.0.0 (OK)
cookiecutter >=1.6.0                  :  1.7.2 (OK)
diff_match_patch >=20181111           :  20200713 (OK)
intervaltree >=3.0.2                  :  3.1.0 (OK)
IPython >=7.6.0                       :  7.27.0 (OK)
jedi >=0.17.2;<0.19.0                 :  0.17.2 (OK)
jellyfish >=0.7                       :  None (OK)
jsonschema >=3.2.0                    :  3.2.0 (OK)
keyring >=17.0.0                      :  23.1.0 (OK)
nbconvert >=4.0                       :  6.1.0 (OK)
numpydoc >=0.6.0                      :  1.1.0 (OK)
parso >=0.7.0;<0.9.0                  :  0.7.0 (OK)
pexpect >=4.4.0                       :  4.8.0 (OK)
pickleshare >=0.4                     :  0.7.5 (OK)
psutil >=5.3                          :  5.8.0 (OK)
pygments >=2.0                        :  2.10.0 (OK)
pylint >=2.5.0                        :  2.9.6 (OK)
pyls_spyder >=0.4.0                   :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0                  :  1.3.0.dev0 (OK)
pylsp_black >=1.0.0                   :  None (OK)
qdarkstyle =3.0.2                     :  3.0.2 (OK)
qstylizer >=0.1.10                    :  0.1.10 (OK)
qtawesome >=1.0.2                     :  1.0.2 (OK)
qtconsole >=5.1.0                     :  5.1.1 (OK)
qtpy >=1.5.0                          :  1.10.0 (OK)
rtree >=0.9.7                         :  0.9.7 (OK)
setuptools >=49.6.0                   :  58.0.4 (OK)
sphinx >=0.6.6                        :  4.2.0 (OK)
spyder_kernels >=2.1.1;<2.2.0         :  2.2.0.dev0 (OK)
textdistance >=4.2.0                  :  4.2.1 (OK)
three_merge >=0.1.1                   :  0.1.1 (OK)
watchdog >=0.10.3                     :  2.1.3 (OK)
xdg >=0.26                            :  0.27 (OK)
zmq >=17                              :  22.2.1 (OK)

# Optional:
cython >=0.21                         :  0.29.24 (OK)
matplotlib >=2.0.0                    :  3.4.2 (OK)
numpy >=1.7                           :  1.21.2 (OK)
pandas >=1.1.1                        :  1.3.3 (OK)
scipy >=0.17.0                        :  1.7.1 (OK)
sympy >=0.7.3                         :  1.8 (OK)

**Comment by steff456** [comment]
Hi @ok97465,

The colors of the console are linked to Spyder's theme, you can change it on `Preferences > Appearance > Syntax Highlighting theme`. You can either choose one from the list of themes, or change any color by clicking in `Edit selected scheme` or even create a new one. 

Hope this helps! Please let me know if there's anything else you need help with

**Comment by green-slowmoe** [comment]
so now that IPython 8 is here, how should one go about homogenizing its colors with the rest of spyder?

**Comment by je-santos** [comment]
On dark mode it's hard to read the folder names when typing`ls`. It's a very dark blue :(

**Comment by shiganai** [comment]
Hi, I'm facing the same issue.
Spyder is a good environment to debug with iPython,
so it will be great if we can customize the color scheme

**Comment by Ellnz** [comment]
I'm also finding it impossible to view the dark blue columns in the console when printing data.

**Comment by findlayjy** [comment]
> Hi @ok97465,
> 
> The colors of the console are linked to Spyder's theme, you can change it on `Preferences > Appearance > Syntax Highlighting theme`. You can either choose one from the list of themes, or change any color by clicking in `Edit selected scheme` or even create a new one.
> 
> Hope this helps! Please let me know if there's anything else you need help with

This doesn't seem to be true for me, running spyder 5.2.2, at least. The iPython theme is frustratingly stubborn, and the dark blue is impossible to read on a dark background.

**Comment by Aster-the-Med-Stu** [comment]
@ccordoba12 No, I think he is referring to running ls in ipython console. The dark blue color is very hard to read. Dunno if ipython console could follow editor's color schematics.
![image](https://github.com/user-attachments/assets/0b584493-eb85-4481-934c-74577c5298e0)

**Comment by ccordoba12** [comment]
Those colors are hard-coded in IPython (the package that powers the console).

@dalthviz, is there any chance we could remap the ANSI colors used for `ls` on our side?

**Comment by dalthviz** [comment]
I think that would mean to do something like the handling of the yellow color that was being done over QtConsole and was reverted over https://github.com/jupyter/qtconsole/pull/611 Just in case, as a workaround for the `ls` color (which strangely seems like doesn't affect Windows 🤔) maybe using the colors magic `NoColor` option could help. Note, however, that using such option will made all the terminal output to be white. To activate that you can run something like `%colors nocolor` and to revert it you can run something like `%colors linux`

**Comment by ccordoba12** [comment]
> I think that would mean to do something like the handling of the yellow color that was being done over QtConsole and was reverted over https://github.com/jupyter/qtconsole/pull/611

Ok, things don't seem so simple as I thought.

These are a couple of resources that could help to solve this problem:

* https://stackoverflow.com/questions/14129278/how-do-i-customize-text-color-in-ipython
* https://stackoverflow.com/questions/47953896/syntax-colors-on-ipython

**Comment by mnmitch** [comment]
This doesn't resolve the issue but I have found a decent work around for the issue:

1. Create a custom scheme with all the defined colors that you prefer (or just copy the values from a theme you already like)
2. Pick and Apply a scheme that has better console colors (such as Obsidian)
3. Edit the selected scheme to match the values you copied in Step 1

I now have the color scheme I prefer in the editor and color scheme in the console that is easy to see with Dark interface theme.
